webapp/components/utils/Notifications.js

```diff
@@ -40,4 +40,140 @@ export const notifications = [
       },
     },
   },
+  {
+    read: false,
+    reason: 'filed_report_on_resource',
+    from: {
+      __typename: 'Report',
+      id: 'reportOnUser',
+      filed: [
+        {
+          reasonCategory: 'discrimination_etc',
+          reasonDescription: 'This user is harassing me with bigoted remarks!',
+          reportedResource: {
+            __typename: 'User',
+            id: 'badWomen',
+            slug: 'mrs.-badwomen',
+            name: 'Mrs. Badwomen',
+          },
+        },
+      ],
+    },
+  },
 ]
+
+export const extractNotificationDataOfCurrentUser = (notification, currentUser) => {
+  const from = notification.from // for readability
+  let user = null
+  let post = null
+  let comment = null
+  let contentExcerpt = null
+  let report = null
+  let isUser = false
+  let isPost = false
+  let isComment = false
+  let isReport = false
+  let reasonTranslationExtention = ''
+  let iconName, iconTooltip, triggerer, title, author, linkName, linkParams, linkHashParam
+
+  // extract data out of the deep structure of db response
+
+  // leave undefined data as default, see above, so later by priority user, comment, post we get easely a clou what it is
+  switch (from.__typename) {
+    case 'Comment':
+      comment = from
+      post = comment.post
+      isComment = true
+      iconName = 'comment'
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-02-01T14:49:02Z" title="Saturday, February 1st 2020, 3:49:02 pm +01:00">Feb 1, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -40,4 +40,117 @@ export const notifications = [
       },
     },
   },
+  {
+    read: false,
+    reason: 'filed_report_on_resource',
+    from: {
+      __typename: 'Report',
+      id: 'reportOnUser',
+      filed: [
+        {
+          reasonCategory: 'discrimination_etc',
+          reasonDescription: 'This user is harassing me with bigoted remarks!',
+          reportedResource: {
+            __typename: 'User',
+            id: 'badWomen',
+            slug: 'mrs.-badwomen',
+            name: 'Mrs. Badwomen',
+          },
+        },
+      ],
+    },
+  },
 ]
+
+export const extractNotificationDataOfCurrentUser = (notification, currentUser) => {
+  const from = notification.from
+  let triggerer
+  const id = from.id
+  const createdAt = notification.createdAt
+  const read = notification.read
+  const reason = notification.reason
+  let title
+  let author
+  let user = null
+  let post = null
+  let comment = null
+  let contentExcerpt = null
+  let report = null
+  let reasonExtention = ''
+
+  if (from.__typename === 'Post') {
+    post = from
+    triggerer = post.author
+  } else if (from.__typename === 'Comment') {
+    comment = from
+    triggerer = comment.author
+    post = comment.post
```


This can be refactored. Something like this:
```js
const { post } = from.comment || from
const { author: violator } = from
```
etc.

It's just so hard to read.

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-02-06T08:24:40Z" title="Thursday, February 6th 2020, 9:24:40 am +01:00">Feb 6, 2020</time>_

---


I tried to code it as clear as possible again! @roschaefer 

I will discuss this with @mattwr18 !

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-02-06T08:24:40Z" title="Thursday, February 6th 2020, 9:24:40 am +01:00">Feb 6, 2020</time>_

---


I tried to code it as clear as possible again! @roschaefer 

I will discuss this with @mattwr18 !